src/components/inspectors/WeekdaySelect.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <div v-if="cycleManager.isWeeklyPeriodSelected()" class="mb-2">
+    <label class="">{{ $t(weekLabel) }}</label>
+    <div>
+      <span
+        v-for="day in weekdays"
+        :key="day.day"
+        class="badge badge-pill weekday mb-1"
+        :class="cycleManager.weekdayStyle(day)"
+        :data-test="`day-${ day.day }`"
+        @click="clickWeekDay(day)"
+      >
+        {{ $t(day.initial) }}
+      </span>
+    </div>
+    <small v-if="repeatOnValidationError" class="text-danger">{{ repeatOnValidationError }}</small>
+    <small class="form-text text-muted">{{ $t(periodicityHelper) }}</small>
+  </div>
+</template>
+
+<script>
+import { DateTime } from 'luxon';
+import CycleManager from '@/components/inspectors/CycleManager';
+
+export default {
+  props: {
+    selectWeekdays: {
+      type: Array,
+      default: () => [],
+    },
+    startDate: {
+      type: String,
+      default: DateTime.local().toUTC().toISO().toString(),
+    },
+    endDate: {
+      type: String,
+      default: null,
+    },
+    ends: {
+      type: String,
+      default: 'never',
+    },
+    times: {
+      type: [Number, String],
+      default: 1,
+    },
+    periodicityValue: {
+      type: String,
+      required: true,
+    },
+    repeat: {
+      type: [String, Number],
+      default: 1,
+    },
+    hasEnds: {
+      type: Boolean,
+      default: true,
+    },
+    weekLabel: {
+      type: String,
+      default: 'Repeat on',
+    },
+    periodicityHelper: {
+      type: String,
+      default: 'Select the day(s) of the week in which to trigger this element',
+    },
+  },
+  data() {
+    return {
+      DateTime,
+      weekdays: [
+        //  ISO week date weekday number, from 1 through 7,
+        //  beginning with Monday and ending with Sunday.
+        {
+          day: 7,
+          initial: 'S',
+          selected: false,
+        },
+        {
+          day: 1,
+          initial: 'M',
+          selected: false,
+        },
+        {
+          day: 2,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 3,
+          initial: 'W',
+          selected: false,
+        },
+        {
+          day: 4,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 5,
+          initial: 'F',
+          selected: false,
+        },
+        {
+          day: 6,
+          initial: 'S',
+          selected: false,
+        },
+      ],
+      cycleManager: new CycleManager(this.startDate, this.repeat, this.periodicityValue, this.selectedWeekdays, this.endDate, this.ends, this.times),
+    };
+  },
+  watch: {
```

Comment:
It seems some of these props/watchers are set up to mirror the value on the cycle manager class. Do you think managing this class from the parent, and passing down the class instead of the individual props is possible / will help?

Reply:
I had trouble trying to pass cycleManager in as a prop. If that is what you were meaning. It often kept old copies of the values. Would need to mob on it.

Reply:
Yeah, that's what I meant. But now that I'm thinking about it, it's not as clear cut as I thought. Changing a value on an instance of that class won't trigger any of Vue's reactivity hooks, and that's good, since it sidesteps the whole unidirectional flow of data between components and should be avoided anyway. That may be what you were experiencing.

To make it work, you'd have to create and pass down a new instance of `CycleManager` each time, which also doesn't sound right. I'm wondering if pulling back a bit and not using a class might be a better approach? Maybe just having a collection of utility functions? Not too sure, but right now this works and what I'm thinking of would require an unknown amount of extra work so let's get this merged.